applications/FluidDynamicsApplication/tests/cpp_tests/test_d_vms_dem_coupled_element.cpp

```diff
@@ -0,0 +1,130 @@
+    ConstitutiveLaw::Pointer pConsLaw(new Newtonian2DLaw());
+    p_properties->SetValue(CONSTITUTIVE_LAW, pConsLaw);
+
+    // Geometry creation
+    model_part.CreateNewNode(1, 0.0, 0.0, 0.0);
+    model_part.CreateNewNode(2, 1.0, 0.1, 0.0);
+    model_part.CreateNewNode(3, 0.0, 1.0, 0.0);
+    model_part.CreateNewNode(4, 1.0, 0.9, 0.0);
+
+    for (ModelPart::NodeIterator it_node=model_part.NodesBegin(); it_node<model_part.NodesEnd(); ++it_node){
+        it_node->AddDof(VELOCITY_X,REACTION_X);
+        it_node->AddDof(VELOCITY_Y,REACTION_Y);
+        it_node->AddDof(VELOCITY_Z,REACTION_Z);
+        it_node->AddDof(PRESSURE,REACTION_WATER_PRESSURE);
+        double& r_fluid_fraction = it_node->FastGetSolutionStepValue(FLUID_FRACTION);
+        r_fluid_fraction = 1.0;
+    }
+
+    std::vector<ModelPart::IndexType> element_nodes {1, 2, 4, 3};
+    model_part.CreateNewElement("DVMSDEMCoupled2D4N", 1, element_nodes, p_properties);
+
+    // Loop starts at 1 because you need one less clone than time steps (JC)
+    for (unsigned int i = 1; i < buffer_size; i++) {
+        model_part.CloneTimeStep(i * delta_time);
+    }
+
+    // Define the nodal values
+    Matrix reference_velocity(4,2);
+    reference_velocity(0,0) = 0.0; reference_velocity(0,1) = 0.1;
+    reference_velocity(1,0) = 0.1; reference_velocity(1,1) = 0.2;
+    reference_velocity(2,0) = 0.2; reference_velocity(2,1) = 0.3;
+    reference_velocity(3,0) = 0.3; reference_velocity(3,1) = 0.4;
+
+
+    Geometry<Node<3>>& r_geometry = model_part.ElementsBegin()->GetGeometry();
+
+
+    for(unsigned int i=0; i<4; i++){
+        r_geometry[i].FastGetSolutionStepValue(PRESSURE)    = 0.0;
+        r_geometry[i].FastGetSolutionStepValue(PRESSURE, 1) = 0.0;
+        for(unsigned int k=0; k<2; k++){
+            r_geometry[i].FastGetSolutionStepValue(VELOCITY)[k]    = reference_velocity(i,k);
+            r_geometry[i].FastGetSolutionStepValue(VELOCITY, 1)[k] = 0.9*reference_velocity(i,k);
+            r_geometry[i].FastGetSolutionStepValue(MESH_VELOCITY)[k]    = 0.0;
+            r_geometry[i].FastGetSolutionStepValue(MESH_VELOCITY, 1)[k] = 0.0;
+        }
+    }
+
+    // RHS and LHS
+    Vector RHS = ZeroVector(12);
+    Matrix LHS = ZeroMatrix(12,12);
+
+    std::vector<double> output = {2.186533804,2.973017643,-0.059632738,-3.422867224,1.8160847,-0.0474755927,-5.326122843,-5.034362124,-0.03999724473,0.2908305509,-6.026365931,-0.05289442457}; // DVMSDEMCoupled2D4N
+
+    for (ModelPart::ElementIterator i = model_part.ElementsBegin(); i != model_part.ElementsEnd(); i++) {
+        const auto& r_process_info = model_part.GetProcessInfo();
+        i->Initialize(r_process_info); // Initialize constitutive law
+
+        const auto& rElem = *i;
+        rElem.Check(r_process_info);
+        i->CalculateLocalVelocityContribution(LHS, RHS, r_process_info);
+        std::cout.precision(10);
+        //std::cout << i->Info() << RHS << std::endl;
```

Comment:
remove comments

Reply:
In this specific case I'd keep it (these are really useful when it is required to update the test values due to a formulation upgrade or correction).

Reply:
I see.